File: libraries/graphql/src/main/java/com/tokopedia/graphql/data/source/GraphqlDataStore.java
package com.tokopedia.graphql.data.source;

import com.tokopedia.graphql.data.model.GraphqlCacheStrategy;
import com.tokopedia.graphql.data.model.GraphqlRequest;
import com.tokopedia.graphql.data.model.GraphqlResponseInternal;

import java.util.List;

import rx.Observable;

/**
 * Use kotlin version
 */
@Deprecated
public interface GraphqlDataStore {
    Observable<GraphqlResponseInternal> getResponse(List<GraphqlRequest> requests, GraphqlCacheStrategy cacheType);
}

File: libraries/composite_lib/graphql/src/main/java/com/tokopedia/graphql/coroutines/data/source/GraphqlDataStore.kt
package com.tokopedia.graphql.coroutines.data.source

import com.tokopedia.graphql.data.model.GraphqlCacheStrategy
import com.tokopedia.graphql.data.model.GraphqlRequest
import com.tokopedia.graphql.data.model.GraphqlResponseInternal

interface GraphqlDataStore{
    suspend fun getResponse(requests: List<GraphqlRequest>, cacheStrategy: GraphqlCacheStrategy): GraphqlResponseInternal
}